main.tf

```diff
@@ -1,31 +1,53 @@
-provider "aws" {
-  region = "eu-west-3"
-}
-
-resource "aws_vpc" "test-vpc" {
+//TODO: rename VPC
+resource "aws_vpc" "nginx_vpc" {
   cidr_block = "10.0.0.0/16"
-
+  
   tags = {
-    Name = "Premium Auto VPC"
+    Name = "Nginx VPC"
   }  
 }
 
 //Application Load Balancer
-resource "aws_lb" "alb" {
-  name = "premium-auto-lb"
+//TODO add listeners and target groups
+
+resource "aws_lb" "app_lb" {
+  name = "nginx-load-balancer"
   load_balancer_type = "application"
-  security_groups = [aws_security_group.sg-lb.id]
+  security_groups = [aws_security_group.sg_lb.id]
   subnets = aws_subnet.public_subnet.*.id
   enable_cross_zone_load_balancing = true
+  
+}
+
+resource "aws_lb_target_group" "group" {
+  name = "terraform-alb-target"
+  port = 80
+  protocol = "HTTP"
+  vpc_id = aws_vpc.nginx_vpc.id
+
 }
 
-resource "aws_launch_configuration" "cfg" {
+resource "aws_lb_listener" "listener_http" {
+  load_balancer_arn = aws_lb.app_lb.arn
+  port = "80"
+  protocol = "HTTP"
 
-  name = "premium_cfg"
+  default_action {
+    target_group_arn = aws_lb_target_group.group.arn
+    type = "forward"
+  }
+  tags = {
+    "Name" = "HTTP Listener"
+  }
+}
+
+resource "aws_launch_configuration" "nginx_config" {
+
+  name = "nginx_config"
   image_id = "ami-0f15e0a4c8d3ee5fe"
-    instance_type = "t2.micro"
-    key_name = "ec2"
-  security_groups = [aws_security_group.sg-lb.id]
+  instance_type = "t2.micro"
```

Comment:
Instance type can be `t3.micro`

Reply:
Changed instance type from t2.micro to t3.micro